### PR TITLE
MLW-874: Implement filters

### PR DESCRIPTION
**Pull request by jomadoye**
screen shot 2018-12-17 at 10 37 52 am

### CODE REVIEW

File: src/screening/ScreeningFilters.jsx [hunk 1]
```diff
@@ -1,50 +1,103 @@
...
-          autoFocus
-          placeholder="search by text"
-          type="text"
-        />           
-        <Glyphicon className="right-icon" glyph="remove-sign" />
-      </span>
-    </div>
-  </div>
-);
+class ScreeningFilters extends React.Component {
```
Comment (jomadoye) [inline src/screening/ScreeningFilters.jsx]: @mogoodrich For the identifier filters, I am checking through all the identifiers on the cardList and using it to populate the filter dropdown, is that correct? I 

Also I checked all thorough the OpenMRS Wiki, I couldn't find the breakdown of a patient identifier.
I.e ABCD-1234-XZY,
I know ABCD, means something and XYZ means something, but I'm not sure what they are. So I had issue coming up with good variable names. I had to call them 
ABDC - `firstIdentifierSearchValue`
1234 - `secondIdentifierSearchValue`
XYZ - `thirdIdentifierSearchValue`
If you can help clarify what they mean, I'll update the code.

File: src/screening/ScreeningFilters.jsx [hunk 2]
```diff
@@ -1,50 +1,103 @@
...
 } from '@openmrs/react-components';
-import {
-  Glyphicon,
-  FormControl,
-} from 'react-bootstrap';
+import { FormControl } from 'react-bootstrap';
+import utils from "../utils";
 import './screeningFilters.css';
 
-let ScreeningFilters = props => (
```
Comment (mogoodrich) [inline src/screening/ScreeningFilters.jsx]: Let's rename this component to "IdentifierFilter"

Also, we can get rid of the complexity of determining the prefix by loading the row data... we can just hard the prefixes...

File: src/screening/ScreeningFilters.jsx (hunk 1 above)
Comment (mogoodrich) [reply src/screening/ScreeningFilters.jsx]: @mseaton  @cioan  do you know what the appropriate names are for these components are, and what potential values are for the prefix and the suffix?

File: src/screening/ScreeningQueue.jsx [hunk 3]
```diff
@@ -34,22 +33,17 @@ let ScreeningQueue = props => {
...
         filters={[...props.filters, patientObjByEncounterTypeFilter(ENCOUNTER_TYPES.CheckInEncounterType.uuid, 'include')]}
         getPatientIdentifiers={utils.getPatientIdentifiers}
         loading={props.updating}
         onMountOtherActionCreators={onMountOtherActionCreators}
-        optionalFilters={PATIENT_IDENTIFIER_FILTERS}
-        optionalFiltersType='or'
         rowData={props.rowData}
         rowSelectedActionCreators={[patientActions.setSelectedPatient, ...props.rowSelectedActionCreators]}
+        ScreeningFilters={ScreeningFilters}
+        searchFilterFields={['name.givenName', 'name.familyName', 'identifiers.0.identifier', 'identifiers.1.identifier', 'identifiers.2.identifier']}
```
Comment (mogoodrich) [inline src/screening/ScreeningQueue.jsx]: A patient can have 0 to n identifiers... is there are way search matcher can handle this?  If not, we can go with this pattern for now, though let's at 'identifiers.4..."  to "identifiers.8...." just to be safe

File: src/screening/vl/VLQueueExpected.jsx [hunk 4]
```diff
@@ -53,11 +52,10 @@ class VLQueueExpected extends React.Component {
...
             () => this.props.dispatch(patientActions.clearSelectedPatient())
-          ]}
-          optionalFilters={PATIENT_IDENTIFIER_FILTERS}
-          optionalFiltersType='or'
-          rowData={Object.values(this.props.patients)}
-          rowSelectedActionCreators={[patientActions.setSelectedPatient, this.redirectToCheckinPageActionCreator]}
+          ] }
+          rowData={ Object.values(this.props.patients) }
+          rowSelectedActionCreators={ [patientActions.setSelectedPatient, this.redirectToCheckinPageActionCreator] }
+          ScreeningFilters={ScreeningFilters}
```
Comment (mogoodrich) [inline src/screening/vl/VLQueueExpected.jsx]: Is it okay to have capitalized props like this if it's a React element?

**Comment by mogoodrich** [comment]
@jomadoye can we make the "ScreeningFilters" component (which should be renamed to "IdentifierFilter") a stand-alone component, instead of having it change the content of the search filter?

File: src/screening/ScreeningFilters.jsx [hunk 5]
```diff
@@ -1,50 +1,103 @@
...
+      second = this.handleUndefinedValues(value, '');
+      this.setState({ secondIdentifierSearchValue: this.handleUndefinedValues(value, '') });
+      searchValue = `${firstIdentifierSearchValue}${second}${thirdIdentifierSearchValue}`;
+    } else if (location === 'third') {
+      third = this.handleUndefinedValues(value, '');
+      this.setState({ thirdIdentifierSearchValue: this.handleUndefinedValues(value, '') });
+      searchValue = `${firstIdentifierSearchValue}${secondIdentifierSearchValue}${third}`;
+    }
+
+    this.props.handleSearchChange(searchValue, 'identifierFilter');
```
Comment (mogoodrich) [inline src/screening/ScreeningFilters.jsx]: We don't want to have the identifier filter change the search filter--they should be separate filters.

**Pull request merged**